marketplaceItems/voiceScopeApp/appResources/appData/voiceScope_app.js

```diff
@@ -0,0 +1,816 @@
+// voiceScope_app.js
+
+//  Created by Mark Brosche on 1-2-2019
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+(function(){
+
+    // *************************************
+    // START UTILITY FUNCTIONS
+    // *************************************
+    
+    var POWER_UP = SoundCache.getSound(Script.resolvePath('./resources/sounds/rezzing.wav'));    
+    var VOLUME = 0.2;
+    var injector;
+    function playSound(sound) {
+        if (sound.downloaded) {
+            if (injector) {   
+                injector.stop();
+            }
+            injector = Audio.playSound(sound, {
+                position: MyAvatar.position,
+                volume: VOLUME,
+                localOnly: true
+            });
+        } 
+    }
+    
+    // *************************************
+    // END UTILITY FUNCTIONS
+    // *************************************
+
+
+    // This function decides how to handle web events from the tablet UI.
+    // used by 'ui' in startup()
+    function onMessage(data) {
+        // EventBridge message from HTML script.
+        switch (data.type) {
+            case "EVENT_BRIDGE_OPEN_MESSAGE":
+                ui.sendToHtml({
+                    type: "buttonStatus",
+                    value: enabled
+                });
+                break;
+            case "TOGGLE_APP":
+                toggleApp();
+                ui.sendToHtml({
+                    type: "buttonStatus",
+                    value: enabled
+                });
+                break;
+        }
+    }
+
+
+    // This function keeps the app running even if you change domains.
+    var DOMAIN_DELAY = 100;
+    function onDomainChange(){
+        // Do not change app status on domain change
+        if (enabled) {
+            Script.setTimeout(function(){
+                startUpdate();
+                addAll();
+            }, DOMAIN_DELAY);
+        }
+    }
+
+
+    // This function loads appui and connects to the needed signals
+    var AppUi = Script.require('appUi');
+    var ui;
+    function startup() {
+        ui = new AppUi({
+            home: Script.resolvePath("./resources/voiceScope_ui.html"),
+            buttonName: "V-SCOPE", // The name of your app
+            graphicsDirectory: Script.resolvePath("./resources/images/"), // Where your button icons are located
+            onMessage: onMessage
+        });       
+        Script.scriptEnding.connect(scriptEnding);
+        Window.domainChanged.connect(onDomainChange);
```

Comment:
looks like there's no `disconnect` signal for this - please add one in `scriptEnding`